### PR TITLE
Preview chaining

### PR DESCRIPTION
**Pull request by hayfield**
Add some linking to the Preview tool.

Built upon #58.

### CODE REVIEW

**Comment by hayfield** [comment]
Ready for after #55 and #58 are merged.

**Comment by dalepotter** [comment]
Nice work, though something doesn't seem to be working with the `stream=True` links.  For example this shows only 50 links on the Previewer, when 6800+ are returned in the [equivalent XML query](http://datastore.iatistandard.org/api/1/access/activity.xml?sector=12220&stream=True).

> Your link: http://datastore.iatistandard.org/api/1/access/activity.csv?sector=12220&stream=True (Preview)

Will take a look at the code when #58 has been merged.

**Comment by hayfield** [comment]
> For example this shows only 50 links on the Previewer, when 6800+ are returned in the equivalent XML query.

That's deliberate - it's a Preview of a subset of the data so this it loads quickly.

**Comment by andylolz** [comment]
Ooh… Exciting! I think this feature could really benefit from user testing. It would be very interesting to know whether users find it useful / confusing / other.

I would say: The previewer shows XML data, whereas the query builder generates CSV. The CSV serializers aren’t even consistent with one another (IATI/IATI-Datastore#275), and they certainly don’t provide a complete representation of the XML data. So in that sense, the CSV is a preview of the XML (and not the other way around!)

Here is a handy tool I made some years ago for previewing CSVs, in case it could be of use:
http://datapipes.okfnlabs.org/html/

Or you could do it the way http://please.treadsoft.ly/query/ and https://octopub.io do it, using [papaparse](http://papaparse.com).

**Comment by hayfield** [comment]
The tool that we maintain to preview IATI data is the Preview tool. At the moment, it allows you to preview XML data, but lacks the functionality to preview CSVs.

While in theory we could bolt something onto the Query Builder, this would go against the rest of our work to make our software components more sustainable and reusable. When we get round to looking at the Preview tool, it may be clear that a way to preview CSVs is atop priority feature. Until then, the available functionality is to preview XML and so that is what is used.

File: templates/index.html [hunk 1]
```diff
@@ -223,7 +223,7 @@ <h1 dir="ltr">IATI Data Store CSV Query Builder (Alpha Version)</h1>
         </form>
 
         <div class="alert alert-success" id="js-query-link" role="alert">
-          <p><strong>Your link:</strong> <a href="http://datastore.iatistandard.org/api/1/access/activity.csv">http://datastore.iatistandard.org/api/1/access/activity.csv</a></p>
+          <p><strong>Your link:</strong> <a id="query-link" href="http://datastore.iatistandard.org/api/1/access/activity.csv">http://datastore.iatistandard.org/api/1/access/activity.csv</a> <span id="preview-link-wrapper">(<a id="preview-link" href="http://preview.iatistandard.org/" target="_blank" rel="noopener noreferrer">Preview</a>)</span></p>
```
Comment (allthatilk) [inline templates/index.html]: Maybe change to `Preview as XML` to avoid user confusion

**Comment by andylolz** [comment]
> While in theory we could bolt something onto the Query Builder, this would go against the rest of our work to make our software components more sustainable and reusable.

Well, I suggested a third party service (datapipes) or third party plugin (papaparse). That’s probably the most sustainable and reusable approach.

**Pull request merged**